### PR TITLE
one.com dns api

### PR DESCRIPTION
**Pull request by diseq**
updated to the new api of one.com
support for multiple domains in one account

### CODE REVIEW

**Comment by Neilpang** [comment]
please check this guide.
https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide

**Comment by diseq** [comment]
checked with the guidelines:

[x] 1. The cloudflare dns api is a recommended reference

[x] 2. The script file name must be myapi.sh: dns_one.sh

[x] 3. The file can be placed in acme.sh/ folder, or in acme.sh/dnsapi/ subfolder.

[x] 4. The file shebang must be sh not bash

[x] 5. There must be 2 functions in your script

[x] 6. Guide for the add function

[x] 7. Guide for the rm function

[x] 8. Please also check this bug to support V2 wildcard cert
       can create same txt record name as long the content is differnt. i.e. wildcard certs working

[x] 9. Update the docs to include your dns api and usage.

[x] 10. Please create a new issue for future bugs

https://github.com/Neilpang/acme.sh/issues/2103

**Comment by Neilpang** [comment]
https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide#9-update-the-docs-to-include-your-dns-api-and-usage

File: dnsapi/dns_one.sh [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+#
+#     only single domain supported atm
+
+dns_one_add() {
+  mysubdomain=$(printf -- "%s" "$1" | rev | cut -d"." -f3- | rev)
+  mydomain=$(printf -- "%s" "$1" | rev | cut -d"." -f1-2 | rev)
+  txtvalue=$2
+
+  # get credentials
+  ONECOM_USER="${ONECOM_USER:-$(_readaccountconf_mutable ONECOM_USER)}"
```
Comment (Neilpang) [inline dnsapi/dns_one.sh]: please rename `ONECOM_USER` to `ONECOM_User`

`ONECOM_PASSWORD` to `ONECOM_Password`

File: dnsapi/dns_one.sh [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  export _H2="Cookie: ${CSRF_G_TOKEN}"
+
+  # Update the IP address for domain entry
+  postdata="{\"type\":\"dns_custom_records\",\"attributes\":{\"priority\":0,\"ttl\":600,\"type\":\"TXT\",\"prefix\":\"$mysubdomain\",\"content\":\"$txtvalue\"}}"
+  _debug postdata "$postdata"
+  response="$(_post "$postdata" "https://www.one.com/admin/api/domains/$mydomain/dns/custom_records" "" "POST" "application/json")"
+  response="$(echo "$response" | _normalizeJson)"
+  _debug response "$response"
+
+  id=$(printf -- "%s" "$response" | sed -n "s/{\"result\":{\"data\":{\"type\":\"dns_custom_records\",\"id\":\"\([^\"]*\)\",\"attributes\":{\"prefix\":\"$mysubdomain\",\"type\":\"TXT\",\"content\":\"$txtvalue\",\"priority\":0,\"ttl\":600}}},\"metadata\":null}/\1/p")
```
Comment (Neilpang) [inline dnsapi/dns_one.sh]: please use `echo` instead of `printf %s` here.
`sed -n` is also too dangerous , please consider replacement.

**Comment by Neilpang** [comment]
please update.

Thanks.

**Comment by Neilpang** [comment]
any update ?

**Pull request merged**